lib/activity_reporter.rb

```diff
@@ -0,0 +1,43 @@
+# generates text for a report on druid count and logs
+class ActivityReporter
+  DRUID_REGEXP = '[[:lower:]]{2}\\d{3}[[:lower:]]{2}\\d{4}'.freeze
+
+  def output(log_files=default_log_files)
+    today = Time.now.to_date.to_s
+    counter = {}
+    puts "*" * 20
+    log_files.each do |file|
+      unless File.exist?(file) && File.size(file) > 0
+        puts "EMPTY or NON-EXISTENT: #{file}"
+        next
+      end
+      extract_druid(file, today, counter)
+      druid_counts = counter.keys.size
+      if druid_counts.zero?
+        puts "No activity #{today}, DRUID count: #{druid_counts}\n"
+      else
+        puts "DRUID count: #{druid_counts} for #{today}\n"
+      end
+    end
+  end
+
+  def extract_druid(file, today, counter)
+    File.readlines(file).each do |line|
+      line =~ /#{today}/ || next
+      line =~ %r{/bundle/ruby|/usr/local/rvm|resque-signals/} && next
+      druid = line.match(/Finished.*(#{DRUID_REGEXP})/) || next
+      counter[druid] = 1
+    end
+  end
+
+  def default_log_files
+    [
+      "sdr_preservationIngestWF_transfer-object.log",
+      "sdr_preservationIngestWF_validate-bag.log",
+      "sdr_preservationIngestWF_verify-apo.log",
+      "sdr_preservationIngestWF_update-moab.log",
+      "sdr_preservationIngestWF_update-catalog.log",
+      "sdr_preservationIngestWF_complete-ingest.log"
```

Comment:
actually, this would be better as. sdr_preservationIngestWF*  if we can glob it, as that allows us to add steps to the workflow without breaking the script.  The trick will be to only do ".log" and avoid ".log1" ".log2" etc. from rotated logs.

This change might imply fixes to other scripts, too?  I will make a ticket for it.

Reply:
#85 created

Reply:
thank you!

Reply:
If that glob has to be reused in multiple scripts, it makes sense to centralize it to a utility class method somewhere.

Reply:
there's already a ticket to make a single script instead of 2 of 'em.